fitstack/models.py

```diff
@@ -417,3 +418,126 @@ def model(self, theta, freq=None, transfer=None):
+    """Fit the stack data to simulation templates."""
+
+    _param_spec = {
+        "offset": {
+            "fixed": False,
+            "value": 0.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": -1.0,
+                "high": 1.0,
+            },
+        },
+        "omega": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": -10.0,
+                "high": 10.0,
+            },
+        },
+        "b_HI": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": -10.0,
+                "high": 10.0,
+            },
+        },
+        # This is the only parameter which is reasonably constrained
+        "b_g": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": 0.95,
+                "high": 1.05,
+            },
+        },
+        "NL": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": -1.0,
+                "high": 2.0,
+            },
+        },
+        "FoGh": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": 0.0,
+                "high": 3.0,
+            },
+        },
+        "FoGg": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": 0.0,
+                "high": 3.0,
+            },
+        },
+        "M_10": {
+            "fixed": False,
+            "value": 1.0,
+            "prior": "Uniform",
+            "kwargs": {
+                "low": 0.0,
+                "high": 5.0,
+            },
+        },
+    }
+
+    # Default base path
+    base_path = (
+        "/project/rpp-chime/chime/stacking/sims/analysis_stacks_comp/psbeam/psbeam/"
+    )
+
+    def __init__(self, *args, **kwargs):
+
+        pattern = (
+            self.base_path
+            + f"dataweight_compderiv*/post/datamask/{self.tracer}/{self.field}/"
+        )
+        factor = 1e3
+
+        self._signal_template = signal.SignalTemplate.load_from_stackfiles(
+            pattern,
+            factor,
+            aliases=dict(shotnoise="M_10"),
+        )
+
+        super().__init__(*args, **kwargs)
+
+    def model(self, theta, freq=None):
+
+        if freq is None:
+            freq = self.freq
+
+        param_dict = {k: v for k, v in zip(self._param_spec.keys(), theta)}
+
+        offset = theta[0]
+
+        model_init = self._signal_template.signal(*param_dict)[[0, 3]]
+
+        model = utils.shift_and_convolve(freq, model_init, offset=offset, transfer=None)
+
+        return model
+
+
+# TODO: this would probably be cleaner if we could pass parameters through to the model
```

Comment:
Yeah I think what I would probably do is make the `base_path` and `pattern` keyword arguments and then create a `model_kwargs` argument to the `run_mcmc` method that is passed to the `Model` constructor.  Then you would just provide the appropriate path for your desired tracer and field in the config file.

Similarly there is a `prior_spec` parameter that is currently passed to the `Model` constructor and will override the defaults set in `_prior_spec` of the class, which will allow us to set different priors for the galaxy bias for the different tracers but use the same underlying class.

Reply:
Sounds good. I didn't want to introduce a whole other bunch of changes in this first iteration, but I agree that it makes sense to make a `model_kwargs` parameter to pass through.

Thanks for the tip on the `prior_spec` I hadn't really noticed to know what that was going to do.

Testing it to see if it works now.... 🤞